### PR TITLE
feat: add tally/copy-after-user-without-chown rule

### PR DESCRIPTION
**Pull request by tinovyatkin**
## Summary

- Adds `tally/copy-after-user-without-chown` rule detecting COPY/ADD without `--chown` after a non-root USER instruction
- Docker's COPY/ADD always creates files as `root:root` regardless of the active USER — a common source of ownership bugs
- Two fix alternatives per violation:
  1. **Add `--chown=<user>`** (preferred) — directly fixes the ownership mismatch
  2. **Move USER past COPY/ADD** — semantic no-op that clarifies intent (USER doesn't affect COPY/ADD ownership)
- Suppressed when a subsequent `RUN chown` already manages ownership of the destination path
- Shared `hasFollowingRunTargetingPath` helper extracted to `instruction_scan.go` for reuse
- Cross-rule composability verified: `--chown` + `--chmod` flags compose correctly on the same COPY instruction

Design doc: `design-docs/34-user-instructions.md` (rule #3)

## Test plan

- [x] Unit tests: 28 cases covering detection, suppression, fixes, edge cases (metadata, USER variants, inherited USER, chown suppression, cross-stage)
- [x] Integration fixture with 8 stages: mixed violation/clean/suppressed cases
- [x] Fix integration tests: basic, ADD, multi-COPY, cross-rule with prefer-copy-chmod
- [x] Shared helper tests: `cmdFileArgCoversPath`, `hasFollowingRunTargetingPath`
- [x] `make test` (5420 tests pass), `make lint` (0 issues), `make cpd` (clean)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added new lint rule tally/copy-after-user-without-chown that warns when COPY/ADD follows a non-root USER without --chown and offers two automatic fix options.

* **Documentation**
  * Added detailed rule documentation, updated the tally rules index, and bumped the README rule count to 46.

* **Tests**
  * Added unit and integration tests plus updated snapshots validating detection, fixes, metadata, and total enabled rules.

### CODE REVIEW

File: internal/rules/tally/copy_after_user_without_chown.go [hunk 1]
```diff
@@ -0,0 +1,428 @@
...
+	}
+
+	if parentFacts := fileFacts.Stage(info.BaseImage.StageIndex); parentFacts != nil {
+		if parentFacts.FinalWorkdir != "" {
+			return parentFacts.FinalWorkdir
+		}
+	}
+
+	return "/"
+}
```
[the remark this answers is not in the document]
Comment (tinovyatkin) [reply internal/rules/tally/copy_after_user_without_chown.go]: ✅ Addressed in b1d09fd. Thanks @coderabbitai\!
